### DISCUSSION TITLE
Clarify how jscache works

### DISCUSSION THREAD

**Discussion by fredrivett** [plausible/analytics#1293] [Anything Else]
In the [`Proxying Plausible through Nginx` docs](https://plausible.io/docs/proxy/guides/nginx#step-1-update-your-nginx-config) it mentions using `jscache`, but when we include it like so:

```
# Only needed if you cache the plausible script. Speeds things up.
proxy_cache_path /var/run/nginx-cache/jscache levels=1:2 keys_zone=jscache:100m inactive=30d  use_temp_path=off max_size=100m;

server {
    listen       8080;
    server_name  localhost;

    #charset koi8-r;
    #access_log  /var/log/nginx/host.access.log  main;

    location = /js/script.js {
        # Change this if you use a different variant of the script
        proxy_pass https://plausible.io/js/plausible.js;

        # Tiny, negligible performance improvement. Very optional.
        proxy_buffering on;

        # Cache the script for 6 hours, as long as plausible.io returns a valid response
        proxy_cache jscache;
        proxy_cache_valid 200 6h;
        proxy_cache_use_stale updating error timeout invalid_header http_500;

        # Optional. Adds a header to tell if you got a cache hit or miss
        add_header X-Cache $upstream_cache_status;
    }

    location = /api/event {
        proxy_pass https://plausible.io/api/event;
        proxy_buffering on;
        proxy_http_version 1.1;

        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host  $host;
    }
}
```

it blows up with the following error:

```
Starting Nginx
2021/09/01 17:26:23 [emerg] 26#26: mkdir() "/var/run/nginx-cache/jscache" failed (2: No such file or directory)
nginx: [emerg] mkdir() "/var/run/nginx-cache/jscache" failed (2: No such file or directory)
```

I've tried googling for `nginx jscache` but little comes up, is there a step I'm missing here to get `jscache` installed?

Thanks folks.

**Comment by ukutaht** [comment]
Hi!

`jscache` in this case is the name you're giving to the cache. Searching for it won't give any results because people give completely different names to their caches. If you want to find more information, I'd suggest searching for `nginx proxy_cache_path` and `nginx proxy_cache`.

I *think* you need to create the `/var/run/nginx-cache` directory on your machine to get it working. Then nginx will be able to create a `/var/run/nginx-cache/jscache` directory where the actual cache data will be stored.

**Reply by fredrivett** [reply]
That's great, thanks so much for the swift and helpful reply here @ukutaht.

I've submitted a PR with a suggested tweak to cover this nuance in the docs here: https://github.com/plausible/docs/pull/120

**Reply by ukutaht** [reply]
Thanks for the PR! Very helpful

**Reply by fredrivett** [reply]
No worries Uku, thanks again for your help!
